src/Security/HypothesisTokenGenerator.php

```diff
@@ -0,0 +1,34 @@
+<?php
+
+namespace eLife\Journal\Security;
+
+use Firebase\JWT\JWT;
+use Symfony\Component\Security\Core\User\UserInterface;
+
+final class HypothesisTokenGenerator
+{
+    const TOKEN_TTL = 10 * 60;
+
+    private $authority;
+    private $clientId;
+    private $clientSecret;
+
+    public function __construct(string $authority, string $clientId, string $clientSecret)
+    {
+        $this->authority = $authority;
+        $this->clientId = $clientId;
+        $this->clientSecret = $clientSecret;
+    }
+
+    public function generate(UserInterface $user) : string
+    {
+        return JWT::encode([
+            'aud' => 'hypothes.is',
+            'iss' => $this->clientId,
+            'sub' => "acct:{$user->getUsername()}@{$this->authority}",
+            'nbf' => time(),
```

Comment:
can `time()` (or `time` as a function) be injected for reproducibility?

Reply:
For testing (which hasn't been done yet)? Symfony can freeze it - https://github.com/elifesciences/journal/blob/develop/test/ViewModel/Factory/FooterFactoryTest.php#L43

Reply:
(Though I have always been more comfortable with clocks as a service rather than namespace-based magic.)

Reply:
This is now tested.